Tribler/community/tunnel/tunnel_community.py

```diff
@@ -172,10 +173,17 @@ def tunnel_data(self, source, data):
         self.community.send_data([Candidate(self.sock_addr, False)], self.circuit_id, ('0.0.0.0', 0), source, data)
 
     def close(self):
+        """
+        Closes the UDP socket if enabled.
+        :return: A deferred that fires once the UDP socket has closed.
+        """
+        defer_list = []
         if self.enabled:
-            self.port.stopListening()
+            defer_list.append(maybeDeferred(self.port.stopListening))
             self.port = None
 
+        return DeferredList(defer_list)
```

Comment:
This doesn't require more branching and can be easily expanded. If the if is false, then the deferred of the DeferredList will fire immediately, else it will wait for the stopListening if needed.